splunk_enterprise_security_tag_assets_and_identities.py

```diff
@@ -0,0 +1,469 @@
+    ################################################################################
+    filter_matching_identities__identities = []
+    for user in collect_users_data___artifact_value:
+        for identity_result in find_identities_result_item_0:
+            if user in identity_result:
+                filter_matching_identities__identities.append(user)
+    phantom.debug(f'Matching identities: {filter_matching_identities__identities}')
+    ################################################################################
+    ## Custom Code End
+    ################################################################################
+
+    phantom.save_run_data(key="filter_matching_identities:identities", value=json.dumps(filter_matching_identities__identities))
+
+    tag_identities(container=container)
+
+    return
+
+
+def users_decision(action=None, success=None, container=None, results=None, handle=None, filtered_artifacts=None, filtered_results=None, custom_function=None, **kwargs):
+    phantom.debug("users_decision() called")
+
+    ################################################################################
+    # Determine if any identities were found.
+    ################################################################################
+
+    # check for 'if' condition 1
+    found_match_1 = phantom.decision(
+        container=container,
+        conditions=[
+            ["collect_users:custom_function_result.data.*.artifact_value", "!=", ""]
+        ])
+
+    # call connected blocks if condition 1 matched
+    if found_match_1:
+        dedup_users(action=action, success=success, container=container, results=results, handle=handle)
+        return
+
+    # check for 'else' condition 2
+    join_collect_hostnames(action=action, success=success, container=container, results=results, handle=handle)
+
+    return
+
+
+def join_collect_hostnames(action=None, success=None, container=None, results=None, handle=None, filtered_artifacts=None, filtered_results=None, custom_function=None, **kwargs):
+    phantom.debug("join_collect_hostnames() called")
+
+    # if the joined function has already been called, do nothing
+    if phantom.get_run_data(key="join_collect_hostnames_called"):
+        return
+
+    # save the state that the joined function has now been called
+    phantom.save_run_data(key="join_collect_hostnames_called", value="collect_hostnames")
+
+    # call connected block "collect_hostnames"
+    collect_hostnames(container=container, handle=handle)
+
+    return
+
+
+def collect_hostnames(action=None, success=None, container=None, results=None, handle=None, filtered_artifacts=None, filtered_results=None, custom_function=None, **kwargs):
+    phantom.debug("collect_hostnames() called")
+
+    id_value = container.get("id", None)
+
+    parameters = []
+
+    parameters.append({
+        "tags": None,
+        "scope": "all",
+        "container": id_value,
+        "data_types": "host, host name, hostname, host_name",
```

Comment:
Fine for now, but eventually we should consider adding internal IP addresses, aws instance IDs, fully qualified domain names, and other sorts of asset names that can show up in splunk.